<!DOCTYPE html>
<html lang="en">
  <style>
    table {
      color: #ED1D99;
    }
    h4 {
      color: #345CB3;
    }
    p {
      color: #345CB3;
      border:2px;
      border-radius: 5px;
      border-style:solid;
      border-color:#ED1D99;
      padding: 1em;
    }
    div {
      text-indent:10px;
    }
  </style>
  <head>
    <meta charset="UTF-8">
    <title>featureCounts and DESeq2</title>
  </head>
  <body>
    <h4>featureCounts and DESeq2</h4>
    <FONT SIZE="2">
      <h4><img src="./logo_destair.png" height="15" /> Tool Rating:</h4>
      <div>
        <table>
          <tr>
            <td>Ease of use:</td>
            <td>&#10026;&#10026;&#10026;&#10026;</td>
          </tr>
          <tr>
            <td>User base:</td>
            <td>&#10026;&#10026;&#10026;</td>
          </tr>
          <tr>
            <td>Sensitivity:</td>
            <td>-</td>
          </tr>
          <tr>
            <td>Speed:</td>
            <td>&#10026;&#10026;&#10026;&#10026;&#10026;</td>
          </tr>
          <tr>
            <td>Configurability:</td>
            <td>&#10026;&#10026;&#10026;&#10026;</td>
          </tr>
        </table>
      </div>
      <h4>Description:</h4>
      <p>
        <b>AWK</b> is a programming language that can be used script
        operations such as parsing, editing, and extract table column values.
        <br /><br ?>
        <b>Infer Experiment</b> calculates the fractions of forward and
        inversely mapped reads onto genomic features with known orientation.
        <br /><br />
        <b>HTSeq-count</b> calculates the number of reads that have been mapped
        against the genomic feature of a reference genome. It is very similar
        to its alternative featureCounts: both require the provided input
        alignment files to be in BAM or SAM format, as well as a genome feature
        file in GTF or GFF format.<br />
        Although these tools return similar results, counts can vary
        considerably in case of paired-ended data. FeatureCounts is faster than
        its counterpart HTSeq-count, and its count table takes into account
        feature overlaps. On the contrary, HTSeq-count is slower and more
        consirvative in terms of counting fragments whose alignment overlaps
        multiple features. Compared to featureCounts, HTSeq-count is older, and
        therefore widely used in the community.
        <br /><br />
        <b>DESeq2</b> assesses the level of gene expression. This tool computes
        ranking and visualization of count data for hypothesis testing under
        multiple sampling conditions.
      </p>
      <h4>Reference:</h4>
      <a target="_blank" href="https://www.gnu.org/software/gawk/manual/gawk.html">AWK</a>
      <a target="_blank" href="https://doi.org/10.1093/bioinformatics/bts356">Infer Experiment</a>
      <a target="_blank" href="https://doi.org/10.1093/bioinformatics/btt656">featureCounts</a>
      <a target="_blank" href="https://doi.org/10.1186/s13059-014-0550-8">DESeq2</a>
    </FONT>
  </body>
</html>
