<!DOCTYPE html>
<html lang="en">
  <style>
    table {
      color: #ED1D99;
    }
    h4 {
      color: #345CB3;
    }
    p {
      color: #345CB3;
      border:2px;
      border-radius: 5px;
      border-style:solid;
      border-color:#ED1D99;
      padding: 1em;
    }
    div {
      text-indent:10px;
    }
  </style>
  <head>
    <meta charset="UTF-8">
    <title>HISAT2</title>
  </head>
  <body>
    <h4>HISAT2</h4>
    <FONT SIZE="2">
      <h4><img src="./logo_destair.png" height="15" /> Tool Rating:</h4>
      <div>
        <table>
          <tr>
            <td>Ease of use:</td>
            <td>&#10026;&#10026;&#10026;&#10026;&#10026;</td>
          </tr>
          <tr>
            <td>User base:</td>
            <td>&#10026;&#10026;&#10026;</td>
          </tr>
          <tr>
            <td>Sensitivity:</td>
            <td>&#10026;&#10026;&#10026;</td>
          </tr>
          <tr>
            <td>Speed:</td>
            <td>&#10026;&#10026;&#10026;&#10026;</td>
          </tr>
          <tr>
            <td>Configurability:</td>
            <td>&#10026;&#10026;&#10026;&#10026;</td>
          </tr>
        </table>
      </div>
      <h4>Description:</h4>
      <p>
<b>HISAT2</b> (Hierarchical Indexing for Spliced Alignment of Transcripts) is a
successor of the HISAT alignmer, and is optimized for memory usage and speed.
This tool is therefore recommended for large data sets.<br>
HISAT2 is able to process single- and paired-end reads in FASTQ-format, and
works with whole-genome, exome, and transcriptome data.
      </p>
      <h4>Reference:</h4>
      <a target="_blank" href="https://doi.org/10.1038/nprot.2016.095">HISAT2</a>
    </FONT>
  </body>
</html>
